src/wallet/spend.rs

```diff
@@ -0,0 +1,411 @@
+
+                let fee_wchange = Amount::from_sat((feerate * vsize_wchange as f64).ceil() as u64);
+
+                #[cfg(feature = "integration-test")]
+                let fee_wchange = Amount::from_sat(1000);
+
+                let remaining_wchange =
+                    if let Some(diff) = total_input_value.checked_sub(total_output_value) {
+                        if let Some(diff) = diff.checked_sub(fee_wchange) {
+                            diff
+                        } else {
+                            return Err(WalletError::InsufficientFund {
+                                available: total_input_value.to_sat(),
+                                required: (total_output_value + fee_wchange).to_sat(),
+                            });
+                        }
+                    } else {
+                        return Err(WalletError::InsufficientFund {
+                            available: total_input_value.to_sat(),
+                            required: (total_output_value + fee_wchange).to_sat(),
+                        });
+                    };
+
+                if remaining_wchange > minimal_nondust {
+                    log::info!(
+                        "Adding change output with {} sats (fee: {} sats)",
+                        remaining_wchange.to_sat(),
+                        fee_wchange.to_sat()
+                    );
+                    tx.output.push(TxOut {
+                        script_pubkey: internal_spk,
+                        value: remaining_wchange,
+                    });
+                } else {
+                    log::info!(
+                        "Remaining change {} sats is below dust threshold. Skipping change output. (fee: {} sats)",
+                        remaining_wchange.to_sat(),
+                        fee_wchange.to_sat()
+                    );
+                }
+            }
+        }
+
+        self.sign_transaction(&mut tx, &mut coins.iter().map(|(_, usi)| usi.clone()))?;
+        let calc_vsize = (tx.base_size() * 4 + total_witness_size).div_ceil(4);
+        let signed_tx_vsize = tx.vsize();
+
+        // As signature size can vary between 71-73 bytes we have a tolerance
+        let tolerance_per_input = 2; // Allow a 2-byte difference per input
+        let total_tolerance = tolerance_per_input * tx.input.len();
+
+        assert!(
+            (calc_vsize as isize - signed_tx_vsize as isize).abs() <= total_tolerance as isize,
+            "Calculated vsize {} didn't match signed tx vsize {} (tolerance: {})",
+            calc_vsize,
+            signed_tx_vsize,
+            total_tolerance
+        );
+
+        log::debug!("Signed Transaction : {:?}", tx.raw_hex());
+        Ok(tx)
+    }
+}
+
+impl OutgoingSwapCoin {
+    pub(crate) fn create_timelock_spend(
+        &self,
+        destination_address: &Address,
+        wallet: &Wallet,
+        feerate: Option<f64>,
+    ) -> Result<Transaction, WalletError> {
```

Comment:
This API is becoming a bit weird. A derivative struct `OutgoingSwapCoin` is taking an arg of a global struct `Wallet`, which is usually not the right pattern.

I think we can simplify this by making the `create_timelock_spen` a wallet API instead of a SwapCoin API. So something like this.

```
impl Wallet {
    fn create_timelock_spend(&self, &outgoingswapcoin, destination, feerate)
```    

This will streamline the API.